transcom/mymove
# B 20940 int

## [B-20940](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20940)

## Summary
B-20940 introduces a disbursement calculation for the SSW, only when a Payment Packet is being created. Please consult the agility backlog item description for the overview of the calculation. Additionally, it is accompanied by a test that checks both methods of calculation, as well as ensuring the Member value cannot be less than 0. This also cleans up an incorrectly named orders value.

> [!NOTE]  
> Until [B-20949](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A997044) is completed, it will be difficult to ensure testing of both these methods of calculation, as the remaining PPM entitlement will always be 0. However, we can almost completely work around this as detailed in the testing instructions.

> [!WARNING]  
> Commit [b626eb1](https://github.com/transcom/mymove/pull/13574/commits/b626eb14b4a8d36e7a3106f5b00d8d03ff664798) is made directly on the int branch to resolve merge conflicts between other SSW features in the server tests only. This solution to the merge conflicts may be cherry-picked to the main PR if necessary.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the app as a service member. 
2. Create a new move.
3. When setting up shipments, create 2 PPMs.
4. The details don't matter too much here, since we don't use the AOA packet. Submit the shipments.
5. Once the moves are submitted, log in as a service counselor and approve the shipments.
6. Log back in as a service member.
7. For both moves, begin the closeout process by finishing the weight ticket you're prompted to do.
8. Once you get the option to add expenses, we'll need to do two different types for our shipments.
9. For one shipment, add an expense with a type of Oil, paid with GTCC. Make the value small, such as $50. 
Then, add another expense type of Storage, with a larger cost, NOT paid with GTCC. 
This triggers our "A" calculation, making GTCC the amount of your oil expense, and allows us to verify member disbursement cannot be less than 0.
10. For the other shipment, add a storage expense not paid with GTCC in a small amount, such as $30. 
Then, add more expensive GTCC expenses. Member disbursement will still be 0 as in the calculation, the member paid storage will cancel out. 
However, this triggers our "B" calculation where the GTCC value will be the amount of the member storage expense.
11. Note: The above instructions assume ppmRemainingEntitlement == 0, per the note above.
12. Once the expenses are configured, submit them.
13. Log back in as a service counselor, approve the expenses.
14. Once the shipment is closed out, download the packet and verify your values against the calculation.

### Backend


## Screenshots
<img width="560" alt="image" src="https://github.com/user-attachments/assets/f44353c7-dbc0-40f3-a48f-b4a2dac2ea84">
